### ISSUE TITLE
downsize seemingly not working on chromium?

### ISSUE THREAD

**Issue by jan-usc** [davidjbradshaw/iframe-resizer#1355]
**Describe the bug**
resizing iframe (up- and downsizing) works perfectly on firefox, but seemingly not in chromium browsers

**To Reproduce**
1. Go to https://www.dav-usc-muenchen.de/iframe
2. select "Alle Touren/Kurse nach Datum'' in the select box "Bitte wählen"
3. the iframe is nicely upsized
4. select "Kanu' in the select box "Bitte wählen"
5. iframe not downsized  (if using chromium browser, eg Chrome, edge, see first screenshot)
5.b iframe downsized accordingly (if using firefox, see scons screenshot)

**Expected behavior**
iframe downsized also when using chromium browser.

**Screenshots**
![grafik](https://github.com/user-attachments/assets/fcb6996b-dd41-4e7f-8d15-665bd0e9eeec)
![grafik](https://github.com/user-attachments/assets/3b7cb0cf-3661-49c5-a330-74d6fd7d3c4b)

**Desktop (please complete the following information):**
 - OS: windows11
 - Browser firefox 131.0.3/Chrome 130.0.6723.92
 - Version see above

**Smartphone (please complete the following information):**
 n/a

**Additional context**

Any help greatly appreciated!

**Comment by davidjbradshaw** [comment]
Hi,

This will turn out to be a CSS issue. _iframe-resizer_ manages to work around most of these, but have a look at this page that details a few ways CSS can trip things up and how to work around them.

https://iframe-resizer.com/content_guidelines/

I’m interested to know how this only affects Chrome, so will take a more detailed look later in the week. It would be very helpful if you could install the latest beta version and turn logging on to enable me to see what is happening.

```sh
npm i @iframe-resizer/child@beta
```

**Comment by jan-usc** [comment]
Hi, 

thx for your quick reply!

I referenced the beta on the child page and activated logging

seems that under Chrome the calculated size doesnt change when the content of the iframe is shrinked (screenshot1),  whereas the size changes if FF (screenshot2)

![grafik](https://github.com/user-attachments/assets/fd2ad089-8e5b-4e13-a180-fb71d1295022)
![grafik](https://github.com/user-attachments/assets/4da339b7-7b33-43d2-ad4e-9a6d98aa44fc)

**Comment by davidjbradshaw** [comment]
Looking at those logs, it appears that you’re loading a new page into the iframe. As part of the page setup _iframe-resizer_ sets the height of both the `body` and `html` elements to **auto**. It appears that something on your page is overriding this value  in Chrome.

The interesting bit, is that after the initial size calculation is done, it appears not to be happy with this and checks it another two times, but your page CSS is still reporting that the body hasn’t downsized.

So my question is how is that height value on `body` being set in Chrome?

**Comment by jan-usc** [comment]
Yes, I do load a new page into the iframe whenever the selection in the select changes. Unfortunately, the webapp within the iframe is based on a rather old framework, the pages are server-side generated whenever an action occurs in the browser. Not a modern single-page client side js app.   

Regarding your question, I'll try to find out, but I need the help of a colleague.

**Comment by yevgeniy2000** [comment]
Hi David,
I'm a colleague of Jan. We found out, that Chrome does not update the size of a div if the content shrinks. It can only increase the `height`. Apperantly it is not possible to change that behaviour using min-height or fit-content.

Instead of using the height of the body-element, we saw that the script adds a empty div (`height=0`) in the end of the iframe.

We found this solution:
To get the correct height, you could use `iframeBodyElement.lastChild.offsetTop`.

**Comment by davidjbradshaw** [comment]
That behaviour is going to be caused by something in your CSS, the solution will be to remove the CSS that causes the issue, rather than adding more CSS to try to overpower the root cause.

If you want to use a a particular element on the page for sizing the iframe, you can add a `data-iframe-size` attribute to that element.

https://iframe-resizer.com/setup/advanced

**Comment by jan-usc** [comment]
I fully agree that we should find the root cause. We tried on a test page w/out any css or js (apart from iframe-resizer), but still doesnt downsize in chrome.

https://tp.dav-usc-muenchen.de/iframe.html

Could you maybe point us to an example page where downsizing works? That could be helpful in finding our root cause...

thx!

**Comment by davidjbradshaw** [comment]
That’s very strange, have you tried the [/examples/html](https://github.com/davidjbradshaw/iframe-resizer/tree/master/example/html) folder in this git project?

**Comment by davidjbradshaw** [comment]
Have you tried another copy/version of Chrome? Just wondering if the issue is a browser plugin?

**Comment by davidjbradshaw** [comment]
I just tried it in Chrome, and don't have the issue. What plugins are you using?

![Screenshot 2024-11-03 at 15 25 52](https://github.com/user-attachments/assets/b9aeac72-095f-4d0f-9415-579eb4da2bff)

**Comment by jan-usc** [comment]
Hi, that's my plugins in chrome:
![grafik](https://github.com/user-attachments/assets/f97aa086-0427-4ae6-af4f-ce3442e686cb)

I also tries with a new clean profile in chrome, also no downsizing. Also in edge without plugins.

in the log you've posted above I can see an upsize from 7834 to 54377, correct?

Do you also see a downsize in the logs when accessing our page? Could you pls select "Kanu" in the select?

![grafik](https://github.com/user-attachments/assets/5bc84d52-93bc-4bdd-8e3a-8ec48024eebb)

I now have a working solution, adding the following div surrounding the whole body content, but still would be nice to find the reason why it doesn't work out of the box.

`<div class="data-iframe-size" data-iframe-size>`

**Comment by davidjbradshaw** [comment]
Sorry for the slow reply, I've been unwell.

Do you still have the broken example online?

**Comment by jan-usc** [comment]
Hi Dave,

yes it's still online:

with
<div class="data-iframe-size" data-iframe-size>

inside body (working as a charm)

https://www.dav-usc-muenchen.de/kurse-touren-veranstaltungen/kalender

without div (downsize not working):
https://www.dav-usc-muenchen.de/iframe


Thanks for your effort!
Jan

**Comment by davidjbradshaw** [comment]
Just had another look at this and I have come to the conclusion that the route cause is that your HTML is invalid, which is then tripping up the browser and making unable to correctly calculate the page heights. I don't think this is really an issue with _iframe-resizer_ and I don't see an easy way for me to detect this.

See the following for a list of errors in your HTML.
https://validator.w3.org/nu/?doc=https%3A%2F%2Ftp.dav-usc-muenchen.de%2Ftp_staging%2FlistFix.do

Given the above I am going to close this ticket, however, if you fix the HTML and still have the same problem, then please feel free to reopen this.
